src/main/java/com/openlattice/postgres/ResultSetAdapters.java

```diff
@@ -747,10 +754,32 @@ public static AppType appType( ResultSet rs ) throws SQLException {
         return data;
     }
 
+    public static Map<Object, Set<Object>> implicitEntityValuesByIdWithVersion(
+            ResultSet rs,
+            Map<UUID, PropertyType> authorizedPropertyTypes,
+            ByteBlobDataManager byteBlobDataManager ) throws SQLException {
+        final Map<Object, Set<Object>> data = new HashMap<>();
+
+        for ( PropertyType propertyType : authorizedPropertyTypes.values() ) {
+            List<?> objects = propertyValue( rs, propertyType );
+
+            if ( objects != null ) {
+                if ( propertyType.getDatatype() == EdmPrimitiveTypeKind.Binary ) {
+                    data.put( propertyType.getId(), new HashSet<>( byteBlobDataManager.getObjects( objects ) ) );
+                } else {
+                    data.put( propertyType.getId(), new HashSet<>( objects ) );
+                }
+            }
+        }
+
+        data.put( ConductorElasticsearchApi.VERSION, ImmutableSet.of( lastWrite( rs ) ) );
```

Comment:
I see. If you are doing this remember property types are random uuids, so you can have uuid constants that can't be generated or you can reserve make an acl key reservation that will never be usable for property types. This avoids the type issues. 

One more thing is that you used lastWrite as opposed to the actual version. Are you using last write instead of version because it's easier to index in elasticsearch?

Reply:
so you mean just use like UUID(0, 0) as version / lastWrite?

I guess we could index all the version numbers instead of just the last write -- I think elasticsearch should be fine to still do range queries on a list of values.

Reply:
Yes

Reply:
I think using UUId(0,0) is more important than switching last write to bersion.